code/TaxonomyTerm.php

```diff
@@ -86,15 +97,57 @@ public function getTaxonomyName()
         return $this->getTaxonomy()->Name;
     }
 
+    /**
+     * Get the type of the top-level ancestor if it is set
+     *
+     * @return string
+     */
+    public function getTaxonomyType()
+    {
+        $taxonomy = $this->getTaxonomy();
+        if ($taxonomy->Type()->exists()) {
```

Comment:
You fixed this by calling ->exists() but calling that on `null` is still an error.